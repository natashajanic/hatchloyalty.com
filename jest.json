{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
