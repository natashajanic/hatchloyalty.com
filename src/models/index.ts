import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
