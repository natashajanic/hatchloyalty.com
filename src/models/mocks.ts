import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
