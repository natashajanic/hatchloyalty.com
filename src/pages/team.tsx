import { graphql, StaticQuery } from 'gatsby'
import * as React from 'react'
import Flex from 'src/components/flex'
import Layout from 'src/components/layout'
import SEO from 'src/components/seo'
import TeamMemberPreview from 'src/components/team-member-preview'
import Text from 'src/components/text'
import Wrapper from 'src/components/wrapper'
import { ITeamMember } from 'src/models'

export interface IPureTeamPageProps {
  data: {
    allMarkdownRemark: {
      teamMembers: Array<{
        name: string
        postCount: number
      }>
    }
    allTeamMembersJson: {
      edges: Array<{
        node: ITeamMember
      }>
    }
  }
}

export const PureTeamPage = (props: IPureTeamPageProps) => {
  const {
    data: {
      allMarkdownRemark: { teamMembers: postCounts },
      allTeamMembersJson: { edges },
    },
  } = props

  const teamMemberElements = edges.map(edge => {
    const memberPostDetails = postCounts.find(
      postDetails => postDetails.name === edge.node.name
    )
    const postCount = memberPostDetails ? memberPostDetails.postCount : 0
    return (
      <TeamMemberPreview
        key={edge.node.name}
        postCount={postCount}
        teamMember={edge.node}
      />
    )
  })

  return (
    <Layout pageStyle="offWhite">
      <SEO title="Hatch Team" />
      <Wrapper my={4}>
        <Text is="h1" fontSize={6} textAlign="center">
          Hatch Team
        </Text>
        <Flex justifyContent="center" my={3}>
          {teamMemberElements}
        </Flex>
      </Wrapper>
    </Layout>
  )
}

class TeamPage extends React.Component<{}, {}> {
  render() {
    const query = graphql`
      {
        allMarkdownRemark(
          limit: 2000
          filter: { fields: { sourceName: { eq: "blog" } } }
        ) {
          teamMembers: group(field: frontmatter___author) {
            name: fieldValue
            postCount: totalCount
          }
        }
        allTeamMembersJson(filter: { active: { eq: true } }) {
          edges {
            node {
              headshot {
                childImageSharp {
                  fluid(maxWidth: 200) {
                    ...GatsbyImageSharpFluid
                  }
                }
              }
              name
              position
            }
          }
        }
      }
    `

    return (
      <StaticQuery
        query={query}
        render={data => <PureTeamPage {...this.props} data={data} />}
      />
    )
  }
}

export default TeamPage
