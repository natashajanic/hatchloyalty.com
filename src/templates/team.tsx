import { Link, graphql } from 'gatsby'
import Img from 'gatsby-image'
import { kebabCase } from 'lodash'
import * as React from 'react'
import system from 'system-components'
import Box from 'src/components/box'
import Layout from 'src/components/layout'
import Panel from 'src/components/panel'
import SEO from 'src/components/seo'
import Text from 'src/components/text'
import Wrapper from 'src/components/wrapper'
import ProfileDefault from 'src/images/profile-default.jpg'
import { IGraphQLImage } from 'src/models'
import Flex from 'src/components/flex'

const TeamMemberCard = system({
  is: Panel,
  flex: 0,
  px: 4,
  py: 4,
  m: 3,
  textAlign: 'center',
})

const TeamMemberImageWrapper = system({
  is: Box,
  borderRadius: 50,
  height: 200,
  mx: 'auto',
  my: 2,
  overflow: 'hidden',
  width: 200,
})

interface ITeamTemplateProps {
  pageContext: {
    basePath: string
    teamMemberNames: string[]
  }
  data: {
    allMarkdownRemark: {
      teamMembers: Array<{
        name: string
        postCount: number
      }>
    }
    allTeamMembersJson: {
      edges: Array<{
        node: {
          headshot: IGraphQLImage
          name: string
          position: string
        }
      }>
    }
  }
}

class TeamTemplate extends React.Component<ITeamTemplateProps, {}> {
  renderTeamMember(teamMemberName: string) {
    const {
      pageContext: { basePath },
      data: {
        allMarkdownRemark: { teamMembers: postCounts },
        allTeamMembersJson: { edges },
      },
    } = this.props

    const matchingEdge = edges.find(edge => edge.node.name === teamMemberName)
    const teamMember = matchingEdge && matchingEdge.node
    if (!teamMember) {
      return null
    }

    const memberPostDetails = postCounts.find(
      postDetails => postDetails.name === teamMemberName
    )
    const postCount = memberPostDetails ? memberPostDetails.postCount : 0

    return (
      <TeamMemberCard key={kebabCase(teamMember.name)}>
        <Link to={`${basePath}/${kebabCase(teamMember.name)}/`}>
          <TeamMemberImageWrapper>
            {teamMember && teamMember.headshot ? (
              <Img fluid={teamMember.headshot.childImageSharp.fluid} />
            ) : (
              <img src={ProfileDefault} alt="" />
            )}
          </TeamMemberImageWrapper>
          <Text is="h2" fontSize={4}>
            {teamMember.name}
          </Text>
          <Text fontSize={3}>
            ({postCount} post{postCount === 1 ? '' : 's'})
          </Text>
        </Link>
      </TeamMemberCard>
    )
  }

  render() {
    const {
      pageContext: { teamMemberNames },
    } = this.props

    const teamMemberElements = teamMemberNames.map(name =>
      this.renderTeamMember(name)
    )
    return (
      <Layout pageStyle="offWhite">
        <SEO title="Hatch Team" />
        <Wrapper my={4}>
          <Text is="h1" fontSize={6} textAlign="center">
            Hatch Team
          </Text>
          <Flex justifyContent="center" my={3}>
            {teamMemberElements}
          </Flex>
        </Wrapper>
      </Layout>
    )
  }
}

export default TeamTemplate

export const pageQuery = graphql`
  {
    allMarkdownRemark(
      limit: 2000
      filter: { fields: { sourceName: { eq: "blog" } } }
    ) {
      teamMembers: group(field: frontmatter___author) {
        name: fieldValue
        postCount: totalCount
      }
    }
    allTeamMembersJson {
      edges {
        node {
          headshot {
            childImageSharp {
              fluid(maxWidth: 200) {
                ...GatsbyImageSharpFluid
              }
            }
          }
          name
          position
        }
      }
    }
  }
`
