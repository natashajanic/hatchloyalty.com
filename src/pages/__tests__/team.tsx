import { shallow, ShallowWrapper } from 'enzyme'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Text from 'src/components/text'
import TeamMemberPreview from 'src/components/team-member-preview'
import { mockTeamMember } from 'src/models/mocks'
import { PureTeamPage, IPureTeamPageProps } from '../team'

describe('TeamPage', () => {
  let props: IPureTeamPageProps
  let mountedComponent: ShallowWrapper<any, any, any>

  beforeEach(() => {
    props = {
      data: {
        allMarkdownRemark: {
          teamMembers: [
            {
              name: 'Sally Smith',
              postCount: 6,
            },
            {
              name: 'John Doe',
              postCount: 15,
            },
          ],
        },
        allTeamMembersJson: {
          edges: [
            {
              node: { ...mockTeamMember, name: 'John Doe' },
            },
            {
              node: { ...mockTeamMember, name: 'Sally Smith' },
            },
            {
              node: { ...mockTeamMember, name: 'George Miller' },
            },
          ],
        },
      },
    }
    mountedComponent = shallow(<PureTeamPage {...props} />)
  })

  describe('render', () => {
    it('displays a page title', async () => {
      expect(mountedComponent.find(Text).length).toEqual(1)
      const titleText = mountedComponent.find(Text).first()
      expect(titleText.text()).toEqual('Hatch Team')
    })

    it('displays a preview for each active team member', async () => {
      expect(mountedComponent.find(TeamMemberPreview).length).toEqual(3)
      const johnDoePreview = mountedComponent.find(TeamMemberPreview).at(0)
      expect(johnDoePreview.prop('teamMember')).toMatchObject({
        ...mockTeamMember,
        name: 'John Doe',
      })
      expect(johnDoePreview.prop('postCount')).toEqual(15)
      const sallySmithPreview = mountedComponent.find(TeamMemberPreview).at(1)
      expect(sallySmithPreview.prop('teamMember')).toMatchObject({
        ...mockTeamMember,
        name: 'Sally Smith',
      })
      expect(sallySmithPreview.prop('postCount')).toEqual(6)
      const georgeMillerPreview = mountedComponent.find(TeamMemberPreview).at(2)
      expect(georgeMillerPreview.prop('teamMember')).toMatchObject({
        ...mockTeamMember,
        name: 'George Miller',
      })
      expect(georgeMillerPreview.prop('postCount')).toEqual(0)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
