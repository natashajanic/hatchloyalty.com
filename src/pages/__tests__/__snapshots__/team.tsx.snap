// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamPage render matches the expected snapshot 1`] = `
<Layout
  pageStyle="offWhite"
>
  <SEO
    title="Hatch Team"
  />
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "maxWidth",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    is="main"
    m="0 auto"
    maxWidth="1024px"
    my={4}
    px={
      Array [
        3,
        4,
      ]
    }
  >
    <Text
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      fontSize={6}
      is="h1"
      lineHeight="1.5"
      m={0}
      textAlign="center"
      textStyle="normal"
    >
      Hatch Team
    </Text>
    <Styled(Component)
      blacklist={
        Array [
          "css",
          "display",
          "flexWrap",
          "alignItems",
          "flexDirection",
          "justifyContent",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "width",
        ]
      }
      display="flex"
      flexWrap="wrap"
      justifyContent="center"
      my={3}
    >
      <TeamMemberPreview
        key="John Doe"
        postCount={15}
        teamMember={
          Object {
            "headshot": Object {
              "childImageSharp": Object {
                "fluid": Object {
                  "aspectRatio": 1,
                  "sizes": "",
                  "src": "test/imgs/john-doe.png",
                  "srcSet": "",
                },
              },
            },
            "name": "John Doe",
            "position": "CEO",
          }
        }
      />
      <TeamMemberPreview
        key="Sally Smith"
        postCount={6}
        teamMember={
          Object {
            "headshot": Object {
              "childImageSharp": Object {
                "fluid": Object {
                  "aspectRatio": 1,
                  "sizes": "",
                  "src": "test/imgs/john-doe.png",
                  "srcSet": "",
                },
              },
            },
            "name": "Sally Smith",
            "position": "CEO",
          }
        }
      />
      <TeamMemberPreview
        key="George Miller"
        postCount={0}
        teamMember={
          Object {
            "headshot": Object {
              "childImageSharp": Object {
                "fluid": Object {
                  "aspectRatio": 1,
                  "sizes": "",
                  "src": "test/imgs/john-doe.png",
                  "srcSet": "",
                },
              },
            },
            "name": "George Miller",
            "position": "CEO",
          }
        }
      />
    </Styled(Component)>
  </Styled(Component)>
</Layout>
`;
